commands/server/start.go

```diff
@@ -2,44 +2,63 @@ package server
 
 import (
 	"flag"
+	"fmt"
+	"github.com/iov-one/weave/errors"
 
-	"github.com/pkg/errors"
-
+	"github.com/iov-one/weave/coin"
 	"github.com/tendermint/tendermint/abci/server"
 	abci "github.com/tendermint/tendermint/abci/types"
 	cmn "github.com/tendermint/tendermint/libs/common"
 	"github.com/tendermint/tendermint/libs/log"
 )
 
 const (
-	flagBind  = "bind"
-	flagDebug = "debug"
+	flagBind   = "bind"
+	flagDebug  = "debug"
+	flagMinFee = "min_fee"
 )
 
-func parseFlags(args []string) (string, bool, error) {
+type Options struct {
+	MinFee coin.Coin
+}
+
+func parseFlags(args []string) (string, bool, *Options, error) {
 	// parse flagBind and return the result
 	var addr string
 	var debug bool
+	var minFeeStr string
+	options := &Options{
+		MinFee: coin.Coin{},
+	}
+
 	startFlags := flag.NewFlagSet("start", flag.ExitOnError)
 	startFlags.StringVar(&addr, flagBind, "tcp://localhost:46658", "address server listens on")
+	startFlags.StringVar(&minFeeStr, flagMinFee, "0 IOV", "minimal anti-spam fee")
 	startFlags.BoolVar(&debug, flagDebug, false, "call stack returned on error")
 	err := startFlags.Parse(args)
-	return addr, debug, err
+
+	if err != nil {
+		return addr, debug, options, err
+	}
+
+	err = options.MinFee.UnmarshalJSON([]byte(fmt.Sprintf(`"%s"`, minFeeStr)))
+
+	return addr, debug, options, err
```

Comment:
Cleaner with return value *Options, err

Reply:
Or maybe it takes in a partially filled in *Options (with home/logger or any defaults) and fills in any other flags as needed, returning only error

Reply:
I'm contemplating to add WithOptions to BaseApp interface to be able to operate on the options easily from within any app.

And I think this will reduce future plumbing to a minimum.

Reply:
Actually, let me see how it turns out, maybe BaseApp is not the best place